.github/workflows/_rust_tests.yml

```diff
@@ -30,25 +42,135 @@ env:
   RUSTUP_MAX_RETRIES: 10
   # RUSTFLAGS: -D warnings
   RUSTDOCFLAGS: -D warnings
-  TEST_ONLY_CHANGED_CRATES: true
   RESTART_POSTGRES: false
 
 jobs:
-  test:
-    name: Test rust crates
-    if: (!cancelled())
+  rust-tests:
+    name: Run rust tests
+    if: |
+      !cancelled() && (inputs.isRust || inputs.isPgIntegration || inputs.isMoveExampleUsedByOthers)
+    timeout-minutes: 90
     runs-on: [self-hosted]
+    env:
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: postgrespw
+      POSTGRES_HOST: localhost
+      POSTGRES_PORT: 5432
+    # we always start the postgres container, even if not all tests need it
+    services:
```

Comment:
Would it make sense to extract and re-use the postgres portion of this file ? 
(into a reusable workflow file ? since it seem to be an exact copy-paste and  probably big enough to be extracted)